### DISCUSSION TITLE
Eager load teams roles in user model

### DISCUSSION THREAD

**Discussion by codebyoul** [spatie/laravel-permission#2164] [Q&A]
Hello,
for not yet logged user there is any best why to set the team_id to eager load the roles with user authentication ?

`protected $with = ['roles', 'permissions'];`

I handle it like this:

`
Route::middleware(['auth:sanctum'])->get('/user', function (Request $request) {
    $user = $request->user();
    setPermissionsTeamId($user->team_id);
    $user->permissions= $user->getPermissionNames();
    $user->roles= $user->getRoleNames();
    return $user;
});

`

Better way to do this ?

**Comment by angeljqv** [comment]
You could overwrite and change functionality
